### PR TITLE
Mentioning the GDExtension icon support

### PR DESCRIPTION
**Pull request by Jackerty**
A node that is loaded through GDExtension can have a custom icon. Icon's path can be added via the `gdextension` file. (See https://github.com/godotengine/godot/pull/75472).

Commit makes a mention of this in the C++ tutorial.

### CODE REVIEW

File: tutorials/scripting/gdextension/gdextension_cpp_example.rst [hunk 1]
```diff
@@ -421,6 +421,22 @@ We're finally ready to run the project:
 
 .. image:: img/gdextension_cpp_animated.gif
 
+Custom editor icon
+------------------
+By default, Godot uses the node icon in the scene dock. The custom icon can be
+added via the ``gdextension`` file. Write a line where the name of the node
+class equals the resource path to the icon's SVG file.
```
Comment (paddy-exe) [inline tutorials/scripting/gdextension/gdextension_cpp_example.rst:428]:
```suggestion
By default, Godot uses the node icon it is inheriting from in the scene dock. The custom icon can be
added via the ``gdextension`` file.  The node's icon is set by reference to its name and resource path
of an SVG file.
```

File: tutorials/scripting/gdextension/gdextension_cpp_example.rst [hunk 2]
```diff
@@ -421,6 +421,22 @@ We're finally ready to run the project:
...
+
+For example:
+
+.. code-block:: none
+
+    [Icon]
+    GDExample = "res://icons/GDExample.svg"
+
+Path points to 16 by 16 pixel SVG image. Read the guide for :ref:`creating icons <doc_editor_icons>`
+for more information.
```
Comment (paddy-exe) [inline tutorials/scripting/gdextension/gdextension_cpp_example.rst:438]:
```suggestion
The path should point to a 16 by 16 pixel SVG image. Read the guide for :ref:`creating icons <doc_editor_icons>`
for more information.
```

**Review by paddy-exe — CHANGES_REQUESTED** [review]
Thanks for your PR! I reworked the sentence structure to clarify a few things and shorten it the important bits. Furthermore, the path should point to a 16 by 16 px image so the icon is scaled properly so I emphasized this. I hope you can agree with my suggestions :)

File: tutorials/scripting/gdextension/gdextension_cpp_example.rst (hunk 1 above)
Comment (Jackerty) [reply tutorials/scripting/gdextension/gdextension_cpp_example.rst:428]: Ummm my editor is not using the icon node it is inheriting from. Is it a bug?

Bug

Comment (paddy-exe) [reply tutorials/scripting/gdextension/gdextension_cpp_example.rst]: In that Case I am wrong. Then leave out the inheriting part.

**Review by paddy-exe — APPROVED** [review]
Looks great! Thanks for the quick updates!

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for contributing and congrats on your first merged PR to Godot's docs! Sorry this took a bit to get merged as I was a bit busy due to real life events and the Valencia sprint :)

**Pull request merged**